v2/pkg/controller/mpi_job_controller.go

```diff
@@ -570,10 +570,24 @@ func (c *MPIJobController) syncHandler(key string) error {
 			}
 		}
 		if launcher == nil {
-			launcher, err = c.kubeClient.BatchV1().Jobs(namespace).Create(context.TODO(), c.newLauncherJob(mpiJob), metav1.CreateOptions{})
-			if err != nil {
-				c.recorder.Eventf(mpiJob, corev1.EventTypeWarning, mpiJobFailedReason, "launcher pod created failed: %v", err)
-				return fmt.Errorf("creating launcher Pod: %w", err)
+			ready := 0
+			if mpiJob.Spec.WaitForWorkers {
+				for _, pod := range worker {
+					for _, c := range pod.Status.Conditions {
+						if c.Type == corev1.PodReady && c.Status == corev1.ConditionTrue {
+							ready++
+						}
+					}
+				}
+			}
+			if !mpiJob.Spec.WaitForWorkers || ready == len(worker) {
```

Comment:
probably worth having a function to improve readability.

```suggestion
			if !mpiJob.Spec.WaitForWorkers || countReadyWorkers(worker) == len(worker) {
```

Reply:
done